R/ahr_blinded.R

```diff
@@ -84,14 +84,14 @@ ahr_blinded <- function(
     hr = c(1, .6),
     ratio = 1) {
   # Input checking
-  if (!is.vector(hr, mode = "numeric") || min(hr) <= 0) {
-    stop("ahr_blinded: hr must be a vector of positive numbers.")
+  if (!is.numeric(hr) || min(hr) <= 0) {
+    stop("'hr' must be a vector of positive numbers.")
   }
-  if (!is.vector(intervals, mode = "numeric") || min(intervals) <= 0) {
-    stop("ahr_blinded: intervals must be a vector of positive numbers.")
```

Comment:
Good question. I deleted the function name in the error message because I thought it would be redundant.

```r
f = function() foo()
foo = function() {
  stop('No!')
}
f()
#> Error in foo() : No!
```

```r
foo = function() {
  stop('foo: No!')
}
f()
#> Error in foo() : foo: No!
```

This relates to a comment I made after https://github.com/Merck/gsDesign2/pull/448#issuecomment-2278528575 That is, the function name will be missing if you `do.call` the function.

```r
foo = function() {
  stop('No!')
}
f = function() do.call(foo, list())
f()
#> Error in (function ()  : No!
```

I think this case is relatively rare. If it does happen, `traceback()` can be helpful to know the chain of function calls.

That said, again I don't have a strong opinion here. In general, my principle is that I'll follow the preference of the project maintainer (i.e. you in this case).

Reply:
Thank you for the clear explanation; I understand your intention. I am open to this. If we choose to remove the function name at the beginning of the error message, we should review the entire gsDesign2 package, as the previous practice of gsDesign2 is to include the function name right at the start (maybe a new issue and PR).

Reply:
Okay, as I'm going through the whole codebase, I'll do this along the way.

Reply:
I searched the repository for how `ahr_blinded()` is used. From the package code, examples, and tests, `ahr_blinded()` is always called directly. In other words, the function itself is never passed as an argument to another function. Thus the fact that an error comes from `ahr_blinded()` should be clear from the default error message, as @yihui noted above.

To demonstrate, I added `stop("there was an error")` to `ahr_blinded()` and generated the error message below:

```R
devtools::load_all()
ahr_blinded(
  surv = survival::Surv(
    time = simtrial::ex2_delayed_effect$month,
    event = simtrial::ex2_delayed_effect$evntd
  ),
  intervals = c(4, 100),
  hr = c(1, .55),
  ratio = 1
)
## Error in ahr_blinded(surv = survival::Surv(time = simtrial::ex2_delayed_effect$month,  : 
##   there was an error
```